### PR TITLE
[XRI3] Updating XRI3 rig to use new "HandPoseDrive" and "PinchInputReader" so to support devices without a Hand Interaction profile

### PR DESCRIPTION
**Pull request by AMollis**
# Overview

 Updating XRI3 rig to use new `HandPoseDrive` and `PinchInputReader` so to support devices without a Hand Interaction profile.

I also have to update the `TapToPlace` solver so it worked with custom `XRInputButtonReaders`.  That is. `PinchInputReader` is a custom `XRInputButtonReaders`, and without this update, the `TapToPlace` behaviour would not work with the new rig.

Also, in this change:

- Disabling some see-it-say-label tests on batch mode.  These tests actually always fail to work in batch mode, since batch mode doesn't have speech services available.
- Changing `PinchInputReader` to cache "valid" action states.  To help improve performance
- Changing `PinchInputReader` with a safer way of unregistering from action events
- Adding an assert in the input test helpers, to catch simultaneous rig creations.

The rig changes are as follows:

1. First screenshot shows the `PinchInputReaders` being added to the rig for the left hand
pr-0

2. Second screenshot shows the `PinchInputReaders` being added to the rig for the right hand.
pr-0-right

3. Next is setting up poke interactor with the `PinchInputReaders` (and explicitly setting the tracked pose driver).
pr-1

4. Setting up the far ray interactor with the `PinchInputReaders` (and explicitly setting the tracked pose driver).
pr-2

5. Setting up the grab interactor with the `PinchInputReaders` (and explicitly setting the tracked pose driver).
pr-3

6. Setting up the gaze pinch interactor with the `PinchInputReaders` (and the tracked pose driver was already set).
pr-4

7. Swapping the old `TrackedPoseDriver` with the extended `HandPoseDriver` tracked pose driver class.
pr-5-pose-driver

8. Moving the `InteractionManager` game object to the top of the rig prefab, so to match the old rig layout.
pr-6-move-interaction-manager-to-top

### CODE REVIEW

File: org.mixedrealitytoolkit.spatialmanipulation/Solvers/TapToPlace.cs [hunk 1]
```diff
@@ -429,10 +431,15 @@ public override void SolverUpdate()
         {
             using (SolverUpdatePerfMarker.Auto())
             {
+                // Stop stop placement if a select action is performed this frame
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.spatialmanipulation/Solvers/TapToPlace.cs:434]:
```suggestion
                // Stop placement if a select action is performed this frame
```

File: org.mixedrealitytoolkit.spatialmanipulation/Solvers/TapToPlace.cs [hunk 2]
```diff
@@ -530,6 +537,27 @@ protected virtual void SetRotation()
             }
         }
 
+        /// <summary>
+        /// Get if an interactor's selcect button was performed this frame.
```
Comment (ms-RistoRK) [inline org.mixedrealitytoolkit.spatialmanipulation/Solvers/TapToPlace.cs:541]:
```suggestion
        /// Get if an interactor's select button was performed this frame.
```

**Comment by shaynie** [comment]
Consistent test failures in the PR build - please review whether the implementation or the tests needs modification.

**Review by shaynie — CHANGES_REQUESTED** [review]
Please address test failures

**Comment by AMollis** [comment]
> Please address test failures

Fixed

File: org.mixedrealitytoolkit.uxcomponents/Tests/Runtime/SeeItSayItCanvasTests.cs [hunk 3]
```diff
@@ -86,6 +86,7 @@ public IEnumerator TestLabelEnabledOnHover()
             yield return hand.Show(Vector3.one);
             yield return RuntimeTestUtilities.WaitForUpdates();
             yield return hand.MoveTo(testButton.transform.position);
+
```
Comment (whebertML) [inline org.mixedrealitytoolkit.uxcomponents/Tests/Runtime/SeeItSayItCanvasTests.cs:89]: This file only seems to have newline changes, intentional?

Comment (ms-RistoRK) [reply org.mixedrealitytoolkit.uxcomponents/Tests/Runtime/SeeItSayItCanvasTests.cs:89]: Improves readability :)

**Review by ms-RistoRK — APPROVED** [review]

**Review by shaynie — APPROVED** [review]

**Pull request merged**